syntax = "proto3";

enum Genre {
  MYSTERY = 0;
  FICTION = 1;
  NON_FICTION = 2;
  ROMANCE = 3;
}

message Book {
  string isbn = 1;
  string title = 2;
  string author = 3;
  Genre genre = 4;
  int32 publishing_year = 5;
}

enum Status {
  AVAILABLE = 0;
  TAKEN = 1;
}

message InventoryItem {
  int32 inventory_number = 1;
  oneof item_type {
    Book book = 2;
  }
  Status status = 3;
}